### ISSUE TITLE
[Utility] Research and make sure we are handling correctly deterministic serialization

### ISSUE THREAD

**Issue by deblasis** [pokt-network/pocket#201]
## Objective

The two JSON below are two different `[]byte` but they represent the same thing. For DDD fans, they represent the same [ValueObject](https://martinfowler.com/bliki/ValueObject.html)

```json
{
  "name": "Satoshi",
  "surname": "Buterin"
}
```

```json
{
  "surname": "Buterin",
  "name": "Satoshi"
}
```

Imagine that we receive tho transactions containing two equivalent `MessageStake` that are different only for the order of their keys in the serialized form.

### What should we do in that case?

- We could be "dumb" and leave the responsibility to the clients.
- We could be "smart" and serialize deterministically. 

## Origin Document

I am creating this issue to log what could be potentially a problem.
The discussion started [here](https://github.com/pokt-network/pocket/pull/177#discussion_r958645385)

TLDR;

- **JSON** payloads need to be sorted lexicographically by key to ensure determinism
- **Protobuf** payloads... not sure if the default implementation is sorted or if it has to be enabled via flag and even in that case, is it guaranteed to stay stable without changes in its implementation aka deterministic? More context [here](https://github.com/pokt-network/pocket/pull/177#discussion_r962355961)

## Goals

- [ ] Research the problem
- [ ] If you can't find anything useful, write tests and verify "scientifically"

## Deliverable

- [ ] Answers :)

## Non-goals / Non-deliverables

- ...

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- **Task specific tests**: `make ...`
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @deblasis 
**Co-Owners**: @Olshansk, @andrewnguyen22
